exporter/keystoneexporter/keystone.go

```diff
@@ -150,3 +176,57 @@ func GetKsGatewayUrl() (string, error) {
 	// "https://ksgateway-${REGION}.${ENV}.netflix.net/REST/v1/stream/${STREAM_NAME}"
 	return fmt.Sprintf("http://ksgateway-%s.%s.netflix.net/REST/v1/stream/%s", region, env, streamName), nil
 }
+
+func getPayload(descriptor *metricspb.MetricDescriptor, series *metricspb.TimeSeries) (*KsPayload, error) {
+	metadata, err := getMetadata(descriptor, series)
+	if err != nil {
+		return nil, err
+	}
+
+	if series.Points == nil {
+		return nil, fmt.Errorf("no points present in metric: %s", descriptor.Name)
+	}
+
+	if len(series.Points) != 1 {
+		return nil, fmt.Errorf("unexpected point count in metric: %s, %d != 1", descriptor.Name, len(series.Points))
+	}
+
+	point := series.Points[0]
+
+	return &KsPayload{
+		Ec2InstanceId: instanceId,
+		NflxStack:     stack,
+		Version:       "v1",
+		Name:          descriptor.Name,
+		Type:          descriptor.Type.String(),
+		Metadata:      metadata,
+		Point: KsPoint{
+			Seconds: point.Timestamp.Seconds,
+			Nanos:   point.Timestamp.Nanos,
+			Value:   point.GetDoubleValue(),
+		},
+	}, nil
+}
+
+func getMetadata(descriptor *metricspb.MetricDescriptor, series *metricspb.TimeSeries) (map[string]string, error) {
+	metadata := make(map[string]string)
+
+	if descriptor.LabelKeys == nil || series.LabelValues == nil {
+		return nil, fmt.Errorf("label keys or values were nil for metric: %s", descriptor.Name)
+	}
+
+	if len(descriptor.LabelKeys) != len(series.LabelValues) {
+		return nil, fmt.Errorf("length of keys and values does not match for metric: %s", descriptor.Name)
+	}
+
+	for i, key := range descriptor.LabelKeys {
+		value := series.LabelValues[i]
+		if value.HasValue {
+			metadata[key.Key] = value.Value
+		} else {
+			metadata[key.Key] = "MISSING"
```

Comment:
What is the tradeoff here, instead to either (a) not set value in map, or (b) use an empty string?

Reply:
This is a very unexpected condition.  I wanted a clear easily queryable signal.